spec/app/ics-028-cross-chain-validation/technical_specification.md

```diff
@@ -28,8 +28,9 @@ Before describing the data structures and sub-protocols of the CCV protocol, we
   (for more details, take a look at the [ABCI documentation](https://docs.tendermint.com/v0.34/spec/abci/abci.html)). 
   In this specification we define the following methods that handle messages that are of particular interest to the CCV protocol:
   - `InitGenesis()` -- Called when the chain is first started, on receiving an `InitChain` message from the consensus engine. 
+    This is also where the application can inform the underlying consensus engine of the initial validator set.
```

Comment:
```suggestion
    This is also where the application must inform the underlying consensus engine of the initial validator set.
```

The ABCI application overall MUST return a valset on init genesis. An individual module within the application may or may not return a valset, but something in application must return valset on initgenesis

Reply:
According to the [ABCI documentation on InitChain](https://docs.tendermint.com/v0.34/spec/abci/abci.html#initchain), 
> If ResponseInitChain.Validators is empty, the initial validator set will be the RequestInitChain.Validators

However, in the SDK this [line](https://github.com/cosmos/cosmos-sdk/blob/7869d38e94d5e691f453880d1be40af60121b85a/types/module/module.go#L338) entails that "Exactly one module must return a non-empty validator set update to correctly initialize the chain"

Reply:
Ahh that is quite interesting. I don't know in practice if `RequestInitChain` does pass in a Validators field. This is a good question to clarify with the tendermint team

Reply:
From my discussion with @sergio-mena:

> `RequestInitChain` contains the set of validators coming from the genesis file. When handling  `RequestInitChain` , the app (e.g., SDK) can override that set with a different one. If the app does not return anything in `ResponseInitChain` , then Tendermint will simply reuse the value it used to populate `RequestInitChain` , which is the one found in the genesis info

> If the genesis info contains an empty validator set, and the app doesn't provide anything in `ResponseInitChain` either, then Tendermint [returns an error](https://github.com/tendermint/tendermint/blob/2183d90d05e1886132f71a4cc68c3048c45261a6/internal/consensus/replay.go#L343). 

> The restriction introduced in the SDK is **_further constraining_** this behavior: the SDK must always override the initial validator set .  But an app using directly ABCI, doesn't necessarily need to override the validator set from genesis

@AdityaSripal  I would suggest to keep the IBC spec consistent with the ABCI spec and not Cosmos SDK, i.e., leave "the application MAY inform". What do you say? 
I also think this should be address in Cosmos SDK to avoid confusion and possible bugs.